### PR TITLE
apprise 1.5.0 (new formula)

### PR DESCRIPTION
**Pull request by jasonlyle88**
Adding a new formula for apprise. This is a python tool generated with the `brew create --python` command and worked without changes.

A test was added to make sure notifications work by creating a custom notifier that just prints to the command line. The output of the command is then checked to look for the expected value.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/a/apprise.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  sha256 "3c581141077a101790ede0cab16fa283ad032a9b2f97dc6d3565fb2d91813fd7"
+  license "BSD-3-Clause"
+
+  depends_on "python@3.11"
+
+  resource "certifi" do
+    url "https://files.pythonhosted.org/packages/98/98/c2ff18671db109c9f10ed27f5ef610ae05b73bd876664139cf95bd1429aa/certifi-2023.7.22.tar.gz"
+    sha256 "539cc1d13202e33ca466e88b2807e29f4c13049d6d87031a3c110744495cb082"
+  end
+
```
Comment (chenrui333) [inline Formula/a/apprise.rb:16]: try with `python-certifi` as dependency

File: Formula/a/apprise.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  resource "idna" do
+    url "https://files.pythonhosted.org/packages/8b/e1/43beb3d38dba6cb420cefa297822eac205a277ab43e5ba5d5c46faf96438/idna-3.4.tar.gz"
+    sha256 "814f528e8dead7d329833b91c5faa87d60bf71824cd12a7530b5526063d02cb4"
+  end
+
+  resource "markdown" do
+    url "https://files.pythonhosted.org/packages/87/2a/62841f4fb1fef5fa015ded48d02401cd95643ca03b6760b29437b62a04a4/Markdown-3.4.4.tar.gz"
+    sha256 "225c6123522495d4119a90b3a3ba31a1e87a70369e03f14799ea9c0d7183a3d6"
+  end
+
```
Comment (chenrui333) [inline Formula/a/apprise.rb:36]: try with `python-markdown` as dependency

File: Formula/a/apprise.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  resource "oauthlib" do
+    url "https://files.pythonhosted.org/packages/6d/fa/fbf4001037904031639e6bfbfc02badfc7e12f137a8afa254df6c4c8a670/oauthlib-3.2.2.tar.gz"
+    sha256 "9859c40929662bec5d64f34d01c99e093149682a3f38915dc0655d5a633dd918"
+  end
+
+  resource "pyyaml" do
+    url "https://files.pythonhosted.org/packages/cd/e5/af35f7ea75cf72f2cd079c95ee16797de7cd71f29ea7c68ae5ce7be1eda0/PyYAML-6.0.1.tar.gz"
+    sha256 "bfdf460b1736c775f2ba9f6a92bca30bc2095067b8a9d77876d1fad6cc3b4a43"
+  end
+
```
Comment (chenrui333) [inline Formula/a/apprise.rb:46]: same

File: Formula/a/apprise.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
+class Apprise < Formula
+  include Language::Python::Virtualenv
+
+  desc "Send notifications from the command-line to popular notification services"
+  homepage "https://pypi.org/project/apprise/"
+  url "https://files.pythonhosted.org/packages/08/b5/07b086d9a984cb296533833eb35ed7553e0eed6f1094a3b434891a7998a7/apprise-1.5.0.tar.gz"
+  sha256 "3c581141077a101790ede0cab16fa283ad032a9b2f97dc6d3565fb2d91813fd7"
+  license "BSD-3-Clause"
+
+  depends_on "python@3.11"
```
Comment (chenrui333) [inline Formula/a/apprise.rb:10]:
```suggestion
  depends_on "python-certifi"
  depends_on "python-markdown"
  depends_on "python@3.11"
  depends_on "pyyaml"
```

**Comment by jasonlyle88** [comment]
Hey @chenrui333 I switched all the resources to dependencies that you mentioned and then re-tested/audited everything. This formula would not work with python-markdown as a dependency, but does work with python-markdown as a resource. The rest of your suggestions worked as dependencies rather than resources (assuming the pipeline agrees!)

**Comment by chenrui333** [comment]
what is the error of using python-markdown as dependency? (this could be some pth issue that we need to wire)

File: Formula/a/apprise.rb (hunk 1 above)
Comment (jasonlyle88) [reply Formula/a/apprise.rb]: Done

**Comment by chenrui333** [comment]
can you also add the resource exclusions into `pypi_formula_mappings.json`, I think we are good to go after that. 👍

File: Formula/a/apprise.rb (hunk 3 above)
Comment (jasonlyle88) [reply Formula/a/apprise.rb]: Done

**Comment by jasonlyle88** [comment]
@chenrui333 When including markdown as a dependency, I get the following error:

```
Traceback (most recent call last):
  File "/usr/local/bin/apprise", line 5, in <module>
    from apprise.cli import main
  File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/__init__.py", line 62, in <module>
    from .Apprise import Apprise
  File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/Apprise.py", line 38, in <module>
    from .conversion import convert_between
  File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/conversion.py", line 34, in <module>
    from markdown import markdown
ModuleNotFoundError: No module named 'markdown'
```

**Comment by chenrui333** [comment]
> @chenrui333 When including markdown as a dependency, I get the following error:
> 
> ```
> Traceback (most recent call last):
>   File "/usr/local/bin/apprise", line 5, in <module>
>     from apprise.cli import main
>   File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/__init__.py", line 62, in <module>
>     from .Apprise import Apprise
>   File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/Apprise.py", line 38, in <module>
>     from .conversion import convert_between
>   File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/conversion.py", line 34, in <module>
>     from markdown import markdown
> ModuleNotFoundError: No module named 'markdown'
> ```

yeah, it is the pth thing

**Comment by jasonlyle88** [comment]
> > @chenrui333 When including markdown as a dependency, I get the following error:
> > ```
> > Traceback (most recent call last):
> >   File "/usr/local/bin/apprise", line 5, in <module>
> >     from apprise.cli import main
> >   File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/__init__.py", line 62, in <module>
> >     from .Apprise import Apprise
> >   File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/Apprise.py", line 38, in <module>
> >     from .conversion import convert_between
> >   File "/usr/local/Cellar/apprise/1.5.0/libexec/lib/python3.11/site-packages/apprise/conversion.py", line 34, in <module>
> >     from markdown import markdown
> > ModuleNotFoundError: No module named 'markdown'
> > ```
> 
> yeah, it is the pth thing

@chenrui333 apologies, this is my first PR for Homebrew. Is it a pathing issue in my PR (and therefore something I need to adjust? Or is it a pathing issue with the `python-markdown` formula that will need to be addressed separately?

**Comment by jasonlyle88** [comment]
... And just saw you open the other PR after my comment, so that answers my question. Thabk you for the help!!!

**Comment by chenrui333** [comment]
np at all.

**Comment by jasonlyle88** [comment]
Hey @chenrui333, just saw that PR #141925 was merged in to master. I know I want to use those changes in my branch so I can use the dependency rather than a resource. However, I'm not quite sure how to do this. Your documenation says 1 commit per formula, 1 formula per commit and not to include a merge commit in my PR. However... I would usually go to my branch and do a `git pull origin master` to pull the updates into my branch... and this would add a merge commit. So how am I supposed to go about this?

Do you recommend I do a `git checkout origin/master -- Formula/p/python-markdown.rb` and then just add that to my commit and push? Or is there a different way you want me to handle this? Thanks for all your help with getting this through so quickly!

**Comment by chenrui333** [comment]
@jasonlyle88 just try squashing all the commits into one and rebase to the latest master. :)

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @jasonlyle88 for your first contribution to homebrew-core! 🎉

**Comment by jasonlyle88** [comment]
Thanks @chenrui333! And thanks for making this an easy contribution!

**Comment by jasonlyle88** [comment]
@chenrui333 or maybe I spoke too soon 🤣. I ran the strict online audit and the new formula audit before submitting (as per documentation I found) and they both completed successfully. Did I miss something else? Is there anything I need to do to fix this?

**Comment by chenrui333** [comment]
that audit failure looks unrelated. merging again.

**Comment by jasonlyle88** [comment]
@chenrui333 same exact failure 😩

**Comment by ZhongRuoyu** [comment]
@jasonlyle88 No worries, the audit failure is unrelated. It's been fixed by #141956.

Thanks for your contribution to Homebrew!

**Pull request merged**